tests/stdlib/metaprogramming/tmacros.nim

```diff
@@ -144,3 +144,13 @@ block: # extractDocCommentsAndRunnables
     
   proc c() {.checkComments("Hello world").} =
     ## Hello world
+
+block some_node_kinds_require_specialize_initialization:
+  ## ensure we can init all the rest as a "negative" test
+  const kindsNotRequiringInit = {low(NimNodeKind)..high(NimNodeKind)} -
+                                  nnkRequireInitKinds
+
+  doAssert kindsNotRequiringInit.len > 0
+
+  for k in kindsNotRequiringInit.items:
+    discard newNimNode(k)
```

Comment:
```suggestion
  static:
    doAssert kindsNotRequiringInit.len > 0

    for k in kindsNotRequiringInit.items:
      discard newNimNode(k)
```
Of course, the compile shouldn't crash when using `newNimNode` outside of a compile-time context. There is a (misplaced) check in `cgen` for this, but it doesn't seem to trigger at the right time.

Reply:
Thanks, the `cgen` crash is what had me confused.